Content/Items/Gear/Weapons/Sword.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System.Linq;
+using Terraria.ID;
+
+namespace FunnyExperience.Content.Items.Gear.Weapons
+{
+	internal class Sword : Gear
+	{
+		public override string Texture => $"{FunnyExperience.ModName}/Assets/Items/Gear/Weapon/Sword/Base";
+	
+		public override void SetDefaults()
+		{
+			Item.damage = 10;
+			Item.width = 40; 
+			Item.height = 40;
+			Item.useStyle = ItemUseStyleID.Swing; 
+			Item.useTime = 20; 
+			Item.useAnimation = 20;
+			Item.autoReuse = true;
+			Item.DamageType = DamageClass.Melee;
+			Item.knockBack = 6;
+			Item.crit = 6;
+			Item.UseSound = SoundID.Item1; 
+		}
+		
+		public override void PostRoll()
+		{
+			//These don't work.... Whai?
+			Item.damage = GetDamage();
+			Item.useTime = GetUseTime();
+		}
+
+		private SwordType FindType()
+		{
+			return Enum.GetValues(typeof(SwordType)).Cast<SwordType>().FirstOrDefault(enumType => Item.Name.Contains(enumType.ToString(), StringComparison.OrdinalIgnoreCase));
+		}
+
+		public int GetDamage()
+		{
+			return FindType() switch
+			{
+				SwordType.Sword => 10,
```

Comment:
For base types I am a bit iffy on hardcoding these as enumerations, If we plan to keep bases soely to stat combinations something data-driven would likely be more viable, and if we plan to do something closer to PoE with different implicit affixes it may be wise to go with an object-oriented approach where we can better define custom behavior for different implicit effects.